### DISCUSSION TITLE
Failing to update the user profile record when creating a user

### DISCUSSION THREAD

**Discussion by bryankmorrow** [pocketbase/pocketbase#817] [Q&A] [answered]
I'm trying to add some values to the user profile when they sign up via a form from my react frontend. However, when I try and follow the documentation examples, I always receive a 404 error on the user.profile.id PATCH request. Here is what the documentation shows for the frontend code:

`// create user
const user = await client.users.create({
    email: 'test@example.com',
    password: '123456',
    passwordConfirm: '123456',
});

// set user profile data
const updatedProfile = await client.records.update('profiles', user.profile.id, {
    name: 'test',
});`

I always receive the following error:

PATCH http://127.0.0.1:8090/api/collections/profiles/records/88t3vvakwrekbkc 404 (Not Found)

But I can verify the profile id (record id) does exist via the Admin console.

Any suggestions on how to resolve this issue?

Thanks,

**Comment by bryankmorrow** [comment]
So looking at this, it looks like this requires Admin authorization.

**Comment by ganigeorgiev** [comment, marked as answer]
@bryankmorrow

That's probably because by default the `profiles` collection can be updated only by admins and the authorized profile owner.

You need to authenticate with the newly created user before making the `clients.records.update()` request OR adjust the `profiles` collection API rules from "Users > Edit profiles collection (settings cogwheel) > API rules".

```js
// create user
const user = await client.users.create({
    email: 'test@example.com',
    password: '123456',
    passwordConfirm: '123456',
});

await client.users.authViaEmail(user.email, "123456");

// set user profile data
const updatedProfile = await client.records.update('profiles', user.profile.id, {
    name: 'test',
});
```

With the users refactoring in the upcoming v0.8 this no longer will be required. You could also check this issue https://github.com/pocketbase/pocketbase/issues/376
